### DISCUSSION TITLE
IOS Open Source Project Contribution

### DISCUSSION THREAD

**Discussion by lexypaul13** [MobileNativeFoundation/discussions#172] [General]
Hello guys, I'm an iOS developer with two years of experience. I am interested in contributing to the development of the MobileNativeFoundation through collaboration and possibly mentorship from skilled iOS engineers. I haven't contributed to an open-source project before; want steps should I take?

**Comment by mhmdzaid** [comment]
hello @lexypaul13 , Contribution to open source projects will come from the apps and frameworks you are using .
Contribution is not necessarily opening PRs . Try to find bugs , read the documentations for each project look for anything that needs to be added or changed.
this is a repo with iOS open source projects @[open-source-ios-apps](https://github.com/dkhamsing/open-source-ios-apps)
you just need to start now 😃 🚀

**Reply by philipphofmann** [reply]
The link to https://github.com/dkhamsing/open-source-ios-apps above is broken. Can you please fix it @mozead1996?

**Comment by danieleformichelli** [comment]
If you are interested in tooling, [Tuist](https://github.com/tuist/tuist/) is a nice open source project to have a look at! 🚀

**Comment by philipphofmann** [comment]
This is another good resource of open source libraries for iOS https://github.com/vsouza/awesome-ios. Just look at something you use yourself and check if the repo is still maintained, meaning check when the last version was released. Some repositories have a `up-for-grabs` label for issues or something similar. You can filter by them to find issues that are easy to fix and don't require a lot of ramp-up. 

You don't necessarily have to write code to contribute to open source. You can check the docs, the readmes, and the contributing files. They often contain typos that are easy to fix, and every maintainer is happy about these fixes. Another idea is to check out the repository and follow the Contributing.md to get started for making contributions. Quite often, they are out of date or don't contain all the information needed, or you find problems along the way. You then can open issues to point out what troubles you had.

**Reply by mkattima1** [reply]
@lexypaul13 , I lead Mobile Platform at Intuit. We have couple of open source projects, with more on the way. Please take a look at https://opensource.intuit.com/app/intuit-open-source/open-source and let me know if you need more details or help with anything. Your feedback will be valuable.
